llmfoundry/utils/config_utils.py

```diff
@@ -656,3 +673,38 @@ def _log_dataset_uri(cfg: Dict[str, Any]) -> None:
         mlflow.log_input(
             mlflow.data.meta_dataset.MetaDataset(source, name=split),
         )
+
+
+def _verify_uc_path(path: str) -> bool:
+    """Verify a UC path exists.
+
+    Args:
+        path (str): UnityCatalog path
+    Returns:
+        (bool): If path exists or not
+    """
+    try:
+        from databricks.sdk import WorkspaceClient
+
+        w = WorkspaceClient()
+    except ImportError:
+        log.warning(
+            'Cannot verify the path of `UCVolumeDatasetSource` because of missing' + \
+            '`databricks-sdk`. Please install `databricks-sdk` via ' + \
+            '`pip install -U databricks-sdk`. This does not block creating ' + \
+            '`UCVolumeDatasetSource`, but your `UCVolumeDatasetSource` might be invalid.',
+        )
+        return False
+    except Exception:
+        log.warning(
+            'Cannot verify the path of `UCVolumeDatasetSource` due to a connection failure ' + \
+            'with Databricks workspace. Please run `mlflow.login()` to log in to Databricks. ' + \
+            'This does not block creating `UCVolumeDatasetSource`, but your ' + \
+            '`UCVolumeDatasetSource` might be invalid.')
+        return False
+
+    try:
+        w.files.get_metadata(path)
+        return True
+    except Exception:
```

Comment:
does this need to be generic exception or can it be more specific?

Reply:
should be Attribute Error, here is traceback:

v.chen@D72F32QJ3J Experiments % python3 debug_oncall.py
Traceback (most recent call last):
  File "/Users/v.chen/Desktop/Experiments/debug_oncall.py", line 28, in <module>
    print(_verify_uc_path('ooga'))
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/v.chen/Desktop/Experiments/debug_oncall.py", line 25, in _verify_uc_path
    w.files.get_metadata(path)
    ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'FilesAPI' object has no attribute 'get_metadata'

Reply:
the error you put is not an attribute error right? its a not found error?